backend/backend/Controllers/UsersController.cs

```diff
@@ -191,36 +191,71 @@ public ActionResult ResetPassword(string newPassword)
             {
                 return NotFound();
             }
+
+            if (string.IsNullOrWhiteSpace(newPassword) || !RegexValidation.IsPasswordValid(newPassword))
+            {
+                return BadRequest();
+            }
             
             user.SetNewPassword(newPassword);
 
             _db.SaveChanges();
 
             return Ok();
         } 
-        [HttpPut, Route("update-credentials")]
-        public ActionResult UpdateCredentials([FromBody] ChangeCredentialsRequest request)
+        [HttpPatch, Route("{id}")]
+        public ActionResult UpdateUser([FromBody] ChangeCredentialsRequest request, [FromRoute] Guid id)
         {
-            if (String.IsNullOrWhiteSpace(request.Email) || !RegexValidation.IsEmailValid(request.Email)) return BadRequest();
-            if (String.IsNullOrWhiteSpace(request.OldPassword) || !RegexValidation.IsPasswordValid(request.OldPassword)) return BadRequest();
-            if (String.IsNullOrWhiteSpace(request.NewPassword) || !RegexValidation.IsPasswordValid(request.NewPassword)) return BadRequest();
-
-            if (request.OldPassword == request.NewPassword) return BadRequest();
-
             var userIdClaim = User.FindFirst(x => x.Type == ClaimTypes.NameIdentifier);
-            if (userIdClaim is null) return NotFound();
-            if (!Guid.TryParse(userIdClaim.Value, out var userId)) return NotFound();
+            if (userIdClaim is null) return NotFound("JWT doesn't correspond to any user");
+            if (!Guid.TryParse(userIdClaim.Value, out var userId)) return NotFound("JWT token is malformed");
+            if (id != userId) return BadRequest("Something wrong has happened");
             var user = _db.Users.FirstOrDefault(x => x.Id == userId);
-            if (user is null) return NotFound();
+            
+            if (user is null) return NotFound("JWT doesn't correspond to any user");
 
-            if (!Hasher.CheckPlaintextAgainstHash(request.OldPassword,user.Password,user.Salt)) return Unauthorized();
-            if (user.Email != request.Email && _db.Users.FirstOrDefault(u => u.Email == request.Email) != null) return Conflict();
+            if (request.Email != user.Email && _db.Users.FirstOrDefault(u => u.Email == request.Email) != null)
+                return Conflict("User with this email already exists");
 
-            user.Email = request.Email;
-            user.SetNewPassword(request.NewPassword);
+            if (string.IsNullOrWhiteSpace(request.Email) && string.IsNullOrWhiteSpace(request.OldPassword)
+                                                         && string.IsNullOrWhiteSpace(request.NewPassword))
+            {
+                return BadRequest("Request is empty");
+            }
```

Comment:
Shouldn't this check be on top of the method? This way we would save unnecessary processing for backend with all those claims etc

Reply:
True that